js/customize-snapshots.js

```diff
@@ -399,6 +588,207 @@
+	 *
+	 * @param {Date} date A Date object.
+	 * @returns {string} A formatted date String.
+	 */
+	component.formatDate = function formatDate( date ) {
+		var formattedDate,
+			yearLength = 4,
+			nonYearLength = 2,
+			monthOffset = 1;
+
+		formattedDate = ( '0000' + date.getFullYear() ).substr( -yearLength, yearLength );
+		formattedDate += '-' + ( '00' + ( date.getMonth() + monthOffset ) ).substr( -nonYearLength, nonYearLength );
+		formattedDate += '-' + ( '00' + date.getDate() ).substr( -nonYearLength, nonYearLength );
+		formattedDate += ' ' + ( '00' + date.getHours() ).substr( -nonYearLength, nonYearLength );
+		formattedDate += ':' + ( '00' + date.getMinutes() ).substr( -nonYearLength, nonYearLength );
+		formattedDate += ':' + ( '00' + date.getSeconds() ).substr( -nonYearLength, nonYearLength );
+
+		return formattedDate;
+	};
+
+	/**
+	 * Populate inputs from the setting value, if none of them are currently focused.
+	 *
+	 * @returns {boolean} Whether the inputs were populated.
+	 */
+	component.populateInputs = function populateInputs() {
+		var parsed;
+
+		if ( component.schedule.inputs.is( ':focus' ) || '0000-00-00 00:00:00' === component.data.publishDate ) {
+			return false;
+		}
+
+		parsed = component.parseDateTime( component.data.publishDate );
+		if ( ! parsed ) {
+			return false;
+		}
+
+		component.schedule.inputs.each( function() {
+			var input = $( this ),
+				fieldName = input.data( 'date-input' );
+
+			if ( ! $( this ).is( 'select' ) && '' === $( this ).val() ) {
+				$( this ).val( parsed[fieldName] );
+			}
+		} );
+		return true;
+	};
+
+	/**
+	 * Populate setting value from the inputs.
+	 *
+	 * @returns {boolean} Whether the date inputs currently represent a valid date.
+	 */
+	component.populateSetting = function populateSetting() {
+		var date = component.getDateFromInputs(),
+			save = $( '#snapshot-save' ),
+			scheduled;
+
+		if ( ! date ) {
+			return false;
+		}
+
+		date.setSeconds( 0 );
+		scheduled = component.formatDate( date ) !== component.data.publishDate;
+
+		if ( save.length ) {
+
+			// Change update button to schedule.
+			if ( component.isFutureDate() ) {
+				save.html( component.data.i18n.scheduleButton );
+			} else {
+				save.html( component.data.i18n.updateButton );
+			}
+
+			if ( scheduled || component.data.dirty ) {
+				save.prop( 'disabled', false );
+			} else {
+				save.prop( 'disabled', true );
+			}
+		}
+
+		component.updateCountdown();
+		component.schedule.template.find( '.reset-time' ).toggle( scheduled );
+
+		return true;
+	};
+
+	/**
+	 * Check if the schedule date is in the future.
+	 *
+	 * @returns {boolean} True if future date.
+	 */
+	component.isFutureDate = function isFutureDate() {
+		var date = component.getDateFromInputs(),
+			millisecondsDivider = 1000,
+			remainingTime;
+
+		if ( ! date ) {
+			return false;
+		}
+
+		remainingTime = component.dateValueOf( date );
```

Comment:
Somewhat of a problem here. `date` is a `Date` object. However, the `component.dateValueOf()` function takes a `string` as its param. It looks like `dateValueOf` needs to take a `Date` as its argument as well?

```diff
diff --git a/js/customize-snapshots.js b/js/customize-snapshots.js
index 40cf91e..6ecd668 100644
--- a/js/customize-snapshots.js
+++ b/js/customize-snapshots.js
@@ -774,19 +774,21 @@
 	/**
 	 * Get the primitive value of a Date object.
 	 *
-	 * @param {string} dateString The post status for the snapshot.
+	 * @param {Date|string} [date] The post status for the snapshot. Defaults to current date.
 	 * @returns {object|string} The primitive value or date object.
 	 */
-	component.dateValueOf = function( dateString ) {
-		var date;
+	component.dateValueOf = function( date ) {
+		var dateObj;
 
-		if ( _.isUndefined( dateString ) ) {
-			date = new Date();
+		if ( _.isUndefined( date ) ) {
+			dateObj = new Date();
+		} else if ( 'string' === typeof date ) {
+			dateObj = new Date( date );
 		} else {
-			date = new Date( dateString );
+			dateObj = date;
 		}
 
-		return date.valueOf();
+		return dateObj.valueOf();
 	};
 
 	component.init();
```